lib/restAPIs/library.js

```diff
@@ -0,0 +1,47 @@
+const assert = require("assert");
+
+const assertParam = require("../util/assertParam");
+const RestUtil = require("../util/RestUtil");
+const buildCommonQuery = require("../util/buildCommonQueryParams");
+
+module.exports = class RestTopic extends RestUtil {
+  _getBase() {
+    return "library";
+  }
+
+  getLibrary(fields = "", outputFlag = "DEFAULT") {
+    const queryParameter = buildCommonQuery(fields);
+
+    return this._makeRequest(
+      `${this._getBase()}`,
+      queryParameter,
+      "GET",
+      null,
+      {
+        outputFlag: outputFlag,
+      }
+    ).then((result) => this.throwIfFailed(result, 200));
+  };
+
+  publishToLibrary(projectID, body){
```

Comment:
Thoughts of `publishDocument` or `publishObject` instead of `publishToLibrary`?

Reply:
Since we can publish either Documents or Dossiers decided to name it 'publisObject'. More generic and who knows what can be published in the future....
Need a bit help with this method. I'm always throwing an exception but status is 204 and the dossier or document gets published.

Reply:
That sounds good! Regarding the successful exception, it's the second parameter to the throwIfFailed() utility method:
https://github.com/tiagosiebler/microstrategy-api-node/blob/master/lib/util/RestUtil.js#L78

Some endpoints return 200 as a success state, while others return 204. If you change 200 to 204 in this method it should resolve after a successful execution.

Reply:
Done.